src/scripts/main.js

```diff
@@ -1,3 +1,257 @@
+
+document.body.append(form);
+
+function showNotification(message, isError = false) {
+  const notification = document.createElement('div');
+
+  notification.setAttribute('data-qa', 'notification');
+  notification.className = isError ? 'error' : 'success';
+  notification.classList.add('notification');
+
+  const title = document.createElement('h2');
+  const description = document.createElement('p');
+
+  title.classList.add('title');
+  title.textContent = isError ? 'Error' : 'Success';
+  description.textContent = message;
+  notification.append(title, description);
+  document.body.append(notification);
+
+  setTimeout(() => {
+    notification.style.display = 'none';
+  }, 5000);
+}
+
+function formatCurrency(value) {
+  return new Intl.NumberFormat('en-US', {
+    style: 'currency',
+    currency: 'USD',
+    minimumFractionDigits: 0,
+    maximumFractionDigits: 0,
+  }).format(value);
+}
+
+submitButton.addEventListener('click', (e) => {
+  e.preventDefault();
+
+  const nameEmployee = form.name.value;
+  const position = form.position.value;
+  const office = form.office.value;
+  const age = parseFloat(form.age.value, 10);
+  const salary = parseFloat(form.salary.value);
+
+  let valid = true;
+  let errorMessage = '';
+
+  if (nameEmployee.length < 4) {
+    valid = false;
+    errorMessage += 'Name must be at least 4 characters long. ';
+  }
+
+  if (age < 18 || age > 90) {
+    valid = false;
+    errorMessage += 'Age must be between 18 and 90. ';
+  }
+
+  if (position.trim().length < 4) {
+    valid = false;
+    errorMessage += 'Position must be at least 4 characters long. ';
+  }
+
+  if (!valid) {
+    showNotification(errorMessage, true);
+
+    return;
+  }
+
+  const newRow = document.createElement('tr');
+
+  const nameCell = document.createElement('td');
+
+  nameCell.textContent = nameEmployee;
+  newRow.appendChild(nameCell);
+
+  const positionCell = document.createElement('td');
+
+  positionCell.textContent = position;
+  newRow.appendChild(positionCell);
+
+  const officeCell = document.createElement('td');
+
+  officeCell.textContent = office;
+  newRow.appendChild(officeCell);
+
+  const ageCell = document.createElement('td');
+
+  ageCell.textContent = age;
+  newRow.appendChild(ageCell);
+
+  const salaryCell = document.createElement('td');
+
+  salaryCell.textContent = formatCurrency(salary);
+  newRow.appendChild(salaryCell);
+
+  tbody.appendChild(newRow);
+
+  form.reset();
+
+  showNotification('Employee successfully added to the table!');
+});
+
+tbody.addEventListener('dblclick', (e) => {
+  const td = e.target.closest('td');
+
+  if (!td) {
+    return;
+  }
+
+  if (td.contentEditable === 'true') {
+    return;
+  }
+
+  td.contentEditable = 'true';
+  td.focus();
+
+  const handleBlur = () => {
+    td.contentEditable = 'false';
+    td.removeEventListener('blur', handleBlur);
+    td.removeEventListener('keydown', handleKeydown);
+  };
+
+  const handleKeydown = (down) => {
```

Comment:
The parameter name `down` is misleading. It should be `event` or `e` to better reflect that it is an event object.